src/plugin.rs

```diff
@@ -20,7 +20,21 @@ pub fn run_plugins(
             .and_then(|cmd| cmd.as_str().map(|s| s.to_owned()))
             .unwrap_or_else(|| format!("yarner-{}", name));
 
-        let args = shlex::split(&command).ok_or(format!("Invalid plugin command {}", command))?;
+        let arguments: Vec<&str> = match config.get("arguments") {
+            None => vec![],
```

Comment:
Isn't there a `.map(..).unwrap_or_default()` in here somewhere as well?

Reply:
Do you mean replacing the match?

Reply:
Yes, as it basically maps `None` to `Vec`'s `Default` impl.

Reply:
Due to a `Option<Result<_, _>>`, I only get it to work like this:

```rust
let arguments: Vec<&str> = config
            .get("arguments")
            .map(|args| {
                args.as_array()
                    .map(|arr| arr.iter().map(|l| l.as_str().unwrap_or_default()).collect())
                    .ok_or("Can't parse array of plugin arguments")
            })
            .unwrap_or_else(|| Ok(vec![]))?;
```

Reply:
I did not compile this, but there is [`transpose`](https://doc.rust-lang.org/stable/std/result/enum.Result.html#method.transpose) to go from `Option<Result<T, E>>` to `Result<Option<T>, E>`, e.g.
```rust
let arguments: Vec<&str> = config
            .get("arguments")
            .map(|args| {
                args.as_array()
                    .map(|arr| arr.iter().map(|l| l.as_str().unwrap_or_default()).collect())
                    .ok_or("Can't parse array of plugin arguments")
            })
            .transpose()?
            .unwrap_or_default();
```